astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1350,7 +1350,7 @@ def fromlineno(self) -> int:
         # lineno is the line number of the first decorator, we want the def
         # statement lineno. Similar to 'ClassDef.fromlineno'
         lineno = self.lineno or 0
-        if self.decorators is not None:
+        if getattr(self, "decorators", None) is not None:
```

Comment:
I'd prefer just setting `decorators` in the ``__init__``. Since it can be `None` anyway this doesn't interfere with the typing of the attribute.